robot-db/src/main/java/repositories/AuthorInput.java

```diff
@@ -0,0 +1,39 @@
+package repositories;
+
+import dao.AuthorDAO;
+import domain.Author;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class tests if Author with given name is in database if not it put him
+ */
+
+@Repository
+public class AuthorInput {
+
+    private Map<String, Author> authors;
+    @Autowired
+    private AuthorDAO authorDAO;
+
+    @PostConstruct
+    private void init () {
+        authors = new HashMap<>();
+        authorDAO.findAll().forEach(author -> authors.put(author.getName(), author));
```

Comment:
Potentially dangerous. If someone ever puts `@Cascade` on your author... you are in world of pain, using DAO on `init` like that.

I'm unsure how to advise cause I don't get why this class was created.